Scaffold backend

nguyennathan opened the pull request:
Scaffolded user backend with Nest CLI

nguyennathan committed:
Scaffold backend

al3xbro commented on user-service/.gitignore:
remove the .gitignore, we already have one in the root folder

al3xbro commented on user-service/README.md:
remove the README as well, we'll put one in the root later

al3xbro commented on user-service/src/app.controller.ts:
remove the default controller, we won't be using it

al3xbro commented on user-service/src/app.service.ts:
remove default service

al3xbro commented on user-service/src/atm/atm.controller.spec.ts:
atms will be using the same endpoint as transactions, as per the hld document, so no need for a separate controller

al3xbro commented on user-service/src/atm/atm.controller.ts:
atms will be using the same endpoint as transactions, as per the hld document, so no need for a separate controller

al3xbro commented on user-service/src/atm/atm.module.ts:
atms will be using the same endpoint as transactions, as per the hld document

al3xbro commented on user-service/src/atm/atm.service.spec.ts:
atms will be using the same endpoint as transactions, as per the hld document. even if they did have separate controllers, we wouldn't have both use the same transaction service

al3xbro commented on user-service/src/atm/atm.service.ts:
atms will be using the same endpoint as transactions, as per the hld document. even if they did have separate controllers, we wouldn't have both use the same transaction service

al3xbro reviewed (changes requested):
great work, looking good. don't forget to remove default starter code. the project organization will change as we add more features and implement more things, but it's good so far.

also name the user-service folder `server`

EthanWen11 committed:
Prisma

created prisma schema, env file contains connection to neon database, added other files to enable Prisma client service

nguyennathan committed:
Merge pull request #5 from al3xbro/prisma

Prisma

nguyennathan committed:
Remove unnecessary files and rename project name to 'server'

al3xbro commented on server/test/app.e2e-spec.ts:
```diff
@@ -1,7 +1,7 @@
 import { Test, TestingModule } from '@nestjs/testing';
 import { INestApplication } from '@nestjs/common';
 import * as request from 'supertest';
-import { AppModule } from './../src/app.module';
```
just ../src/ is fine

al3xbro reviewed (approved):
looking great
